laravel-app/app/Http/Controllers/Auth/RegisterController.php

```diff
@@ -46,39 +47,89 @@ public function __construct()
 
     /**
      * Show login form page.
-     *
-     * @return View
      */
-    public function showRegistrationForm()
+    public function showRegistrationForm(): view
     {
         return view('auth.register');
     }
 
+    /**
+     * Show resend page
+     */
+    public function resendPage(): view
+    {
+        return view('auth.resend');
+    }
+
     /**
      * Create a new user instance after a valid registration.
-     *
-     * @param  CreateUserRequest  $request
-     * @return View
      */
-    public function register(CreateUserRequest $request): View
+    public function register(CreateUserRequest $request): RedirectResponse
     {
         $validatedData = $request->validated();
 
         $user =  User::create([
             'email' => $validatedData['email'],
             'password' => Hash::make($validatedData['password']),
             'username' => $validatedData['username'],
+            'email_verification_code' => Str::random(40),
         ]);
 
         $user->information()->create([
             'first_name' => $validatedData['first_name'],
             'last_name' => $validatedData['last_name'],
         ]);
```

Comment:
Why is this being exposed? create a service layer for this!

Reply:
Yes sir 🙇